src/execution/sql/ddl_executors.cpp

```diff
@@ -156,4 +171,243 @@ bool DDLExecutors::CreateIndex(const common::ManagedPointer<catalog::CatalogAcce
+      const auto &table_col = schema.GetColumn(unique_col);
+      if (table_col.Type() == type::TypeId::VARCHAR || table_col.Type() == type::TypeId::VARBINARY) {
+        key_cols.emplace_back(table_col.Name(), table_col.Type(), table_col.MaxVarlenSize(), table_col.Nullable(),
+                              parser::ColumnValueExpression(connection_db, table, table_col.Oid()));
+
+      } else {
+        key_cols.emplace_back(table_col.Name(), table_col.Type(), table_col.Nullable(),
+                              parser::ColumnValueExpression(connection_db, table, table_col.Oid()));
+      }
+    }
+    catalog::IndexSchema index_schema(key_cols, storage::index::IndexType::BWTREE, true, false, false, true);
+
+    // Create the index, and use its return value as overall success result
+    catalog::index_oid_t index_oid = CreateIndexForConstraints(accessor, plan_node->GetNamespaceOid(),
+                                                               unique_constraint.constraint_name_, table, index_schema);
+    if (index_oid == catalog::INVALID_INDEX_OID) {
+      return false;
+    }
+    unique_cols.clear();
+    unique_cols.reserve(unique_constraint.unique_cols_.size());
+    for (const auto &col_name : unique_constraint.unique_cols_) {
+      unique_cols.push_back(schema.GetColumn(col_name).Oid());
+    }
+    constraint_oid =
+        accessor->CreateUNIQUEConstraints(ns, table, unique_constraint.constraint_name_, index_oid, unique_cols);
+    if (constraint_oid == catalog::INVALID_CONSTRAINT_OID) {
+      // Catalog wasn't able to proceed, txn must now abort
+      return false;
+    }
+  }
+  return true;
+}
+bool DDLExecutors::CreateCheckConstraintsAndIndices(
+    const common::ManagedPointer<catalog::CatalogAccessor> accessor, const terrier::catalog::Schema &schema,
+    const catalog::table_oid_t table, const common::ManagedPointer<planner::CreateTablePlanNode> plan_node,
+    const catalog::db_oid_t connection_db) {
+  // catalog::constraint_oid_t constraint_oid;
+  catalog::namespace_oid_t ns = plan_node->GetNamespaceOid();
+  return (ns != catalog::INVALID_NAMESPACE_OID);
+}
+bool DDLExecutors::CreateExclusionConstraintsAndIndices(
+    const common::ManagedPointer<catalog::CatalogAccessor> accessor, const terrier::catalog::Schema &schema,
+    const catalog::table_oid_t table, const common::ManagedPointer<planner::CreateTablePlanNode> plan_node,
+    const catalog::db_oid_t connection_db) {
+  // catalog::constraint_oid_t constraint_oid;
+  catalog::namespace_oid_t ns = plan_node->GetNamespaceOid();
+  return (ns != catalog::INVALID_NAMESPACE_OID);
+}
+bool DDLExecutors::CreateNotNullConstraints(const common::ManagedPointer<catalog::CatalogAccessor> accessor,
+                                            const terrier::catalog::Schema &schema, const catalog::table_oid_t table,
+                                            const common::ManagedPointer<planner::CreateTablePlanNode> plan_node,
```

Comment:
CreateExclusionConstraintsAndIndices, CreateNotNullConstraints and CreateTriggerConstraints seem to be doing the same thing right now. Also, they are all being called from CreateTableExecutor consecutively. 
Are these functions supposed to contain other logics that are not implemented yet?